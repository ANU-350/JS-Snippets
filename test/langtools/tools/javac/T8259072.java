/*
 * @test /nodynamiccopyright/
 * @bug 8259072
 * @summary access flags inside local class are allowed for source levels previous to 16
 * @compile/fail/ref=T8259072.out -XDrawDiagnostics -source 15 T8259072.java
 */

// Add all three variants + public, protected, or private
public class T8259072 {
    private void PriA1() {
        class Foo {
            private class Test {}
        }
    }

    class PriA2 {
        public PriA2 check() {
            return new PriA2() { private class STRENGTH{}; };
        }
    }

    class PriA3 {
        Object o = new Object() { private class STRENGTH{}; };
    }

    private void PubA1() {
        class Foo {
            public class Test {}
        }
    }

    class PubA2 {
        public PubA2 check() {
            return new PubA2() { public class STRENGTH{}; };
        }
    }

    class PubA3 {
        Object o = new Object() { public class STRENGTH{}; };
    }

    private void ProA1() {
        class Foo {
            protected class Test {}
        }
    }

    class ProA2 {
        public ProA2 check() {
            return new ProA2() { protected class STRENGTH{}; };
        }
    }

    class ProA3 {
        Object o = new Object() { protected class STRENGTH{}; };
    }
}