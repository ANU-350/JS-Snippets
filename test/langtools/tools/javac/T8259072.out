- compiler.warn.source.no.system.modules.path: 15
T8259072.java:12:21: compiler.err.mod.not.allowed.here: private
T8259072.java:18:42: compiler.err.mod.not.allowed.here: private
T8259072.java:23:43: compiler.err.mod.not.allowed.here: private
T8259072.java:28:20: compiler.err.mod.not.allowed.here: public
T8259072.java:34:41: compiler.err.mod.not.allowed.here: public
T8259072.java:39:42: compiler.err.mod.not.allowed.here: public
T8259072.java:44:23: compiler.err.mod.not.allowed.here: protected
T8259072.java:50:44: compiler.err.mod.not.allowed.here: protected
T8259072.java:55:45: compiler.err.mod.not.allowed.here: protected
9 errors
1 warning